superbuild/CMakeLists.txt

```diff
@@ -37,6 +37,13 @@ set(CCACHE_DIR "default" CACHE STRING "ccache cache directory")
 set(PREBUILT_STAGE2 "NO" CACHE STRING "Provide pre-built stage2 archive to skip pre ispc-stage2 jobs")
 set(PREBUILT_STAGE2_PATH "NO" CACHE STRING "Provide pre-built stage2 toolchain path")
 set(STAGE2_TOOLCHAIN_INSTALL_PREFIX "NO" CACHE STRING "Path to install stage2 Clang/LLVM to")
+set(JOBS "" CACHE STRING "Set the number of jobs for internal build commands (ninja)")
+set(BUILD_TYPE "" CACHE STRING "Set the build type for builds (overrides per-component build type switches)")
```

Comment:
I prefer to have default `Release` value for `BUILD_TYPE` to preserve old behavior.

Reply:
Was `BUILD_TYPE` already in use before this patch? I set it to default to empty such that if it's defined to anything by the user, the subcomponent build types which default to Release are overridden. From what I understand the default behavior is preserved.

Reply:
So, it was `Release` and now it is an empty. It is not clear for me why it is same.

Reply:
I think there's some confusion here. As far as I understand, there was no `BUILD_TYPE` variable in the cmake previously, so it would have been unset/empty before this. Are you referring to `CMAKE_BUILD_TYPE`?

Reply:
One idea would be to add `set(CMAKE_BUILD_TYPE $BULD_TYPE)` in the `if (BUILD_TYPE)` section to make sure they match.

Reply:
I sounds like manually defined `BUILD_TYPE` tries to replace `CMAKE_BUILD_TYPE`. I.e. why `CMAKE_BUILD_TYPE` doesn't work for this purpose? It creates confusion and unnecessary complication of the interplay of the two. Do we define what the behavior supposed to be in case both are defined and contradict?

Reply:
We can use `CMAKE_BUILD_TYPE` instead and remove `BUILD_TYPE`. The reason it didn't do what its designed to do is because we were explicitly setting it to Release for each sub-build. The per-component setters allow us to control them separately.

To clear up the implementation, we simply need to rename `BUILD_TYPE` in this patch to `CMAKE_BUILD_TYPE`. That way, if the user sets it, the entire build is of the type requested, but the user can specify type for individual components too.

Reply:
It looks good to me now.